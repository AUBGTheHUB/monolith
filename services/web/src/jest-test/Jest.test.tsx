import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import Link from './Jest';

it('changes the class when hovered', () => {
    const component = renderer.create(<Link />);

    let tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseEnter();
    });
    // re-rendering
    tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseLeave();
    });
    // re-rendering
    tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
    //expect(1).toBeGreaterThan(19);
});
