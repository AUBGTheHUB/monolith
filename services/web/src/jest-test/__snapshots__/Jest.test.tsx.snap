// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Link
</a>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Link
</a>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Link
</a>
`;
